neural_compressor/adaptor/torch_utils/smooth_quant.py

```diff
@@ -764,44 +791,124 @@ def _update_scales_for_auto(self, absorb_scales, weight_scales):
                 weight_scale = self._reshape_scale_for_weight(layer, weight_scale)
                 layer.update_scale(input_scale, weight_scale)  ##FIXME
 
+    def _add_blockwise_observer(self, block_modules):
+        """
+        :param block_modules: the block modules which the observer will insert to
+        :return:
+        """
```

Comment:
This file exceeds 800 lines and it would be beneficial to split it into multiple files in the future, especially since we anticipate adding new features. Large files are more prone to coding bugs, and splitting the code into smaller, more manageable files can help mitigate this issue. @n1ck-guo @xin3he